### PR TITLE
Added infos, clarified some details about omcid patching

### PR DESCRIPTION
**Pull request by gionag**

### CODE REVIEW

File: _ont/ont-fs-com-gpon-onu-stick-with-mac.md [hunk 1]
```diff
@@ -144,20 +144,32 @@ so you need to then modify the binary with the following hex patch which removes
 
 ```
 
-To make it easier you can use the following base64:
+Before proceeding, make sure your `/opt/lantiq/bin/omcid` has the correct md5 checksum `7e97163e24c9cb39439589c65b438168`. If that checks, you can proceed building the patched file:
```
Comment (simonebortolin) [inline _ont/ont-fs-com-gpon-onu-stick-with-mac.md:147]: Move this message into a alert box and add the md5 command

File: _ont/ont-fs-com-gpon-onu-stick-with-mac.md [hunk 2]
```diff
@@ -144,20 +144,32 @@ so you need to then modify the binary with the following hex patch which removes
...
+
+This is the patch, encoded in base64
 ```
 QlNESUZGNDA1AAAAAAAAAD4AAAAAAAAA2C8JAAAAAABCWmg5MUFZJlNZYqnvBwAACFBSQWAAAMAA
 AAgAQCAAMQwIIwjImgDOdMvi7kinChIMVT3g4EJaaDkxQVkmU1lrJSbUAACFTAjAACAAAAiCAAAI
 IABQYAFKQ01INxUgd6Soj2JURm8pUR8XckU4UJBrJSbUQlpoORdyRThQkAAAAAA=
 ```
+Save it on your computer (not on the stick) as `omcid_patch.base64`, then run:
+```
```
Comment (simonebortolin) [inline _ont/ont-fs-com-gpon-onu-stick-with-mac.md:156]:
```suggestion
```sh
```

File: _ont/ont-fs-com-gpon-onu-stick-with-mac.md [hunk 3]
```diff
@@ -144,20 +144,32 @@ so you need to then modify the binary with the following hex patch which removes
...
+base64 -d omcid_patch.base64 > omcid.bspatch
+bspatch <your_original_omcid> omcid omcid.bspatch
+```
+(if you don't have bspatch installed, most distributions includes it in bsdiff package)
```
Comment (simonebortolin) [inline _ont/ont-fs-com-gpon-onu-stick-with-mac.md:160]: Use alertbox

File: _ont/ont-fs-com-gpon-onu-stick-with-mac.md [hunk 4]
```diff
@@ -144,20 +144,32 @@ so you need to then modify the binary with the following hex patch which removes
...
+
+After patching the resulting patched `omcid` should have an md5 checksum of `525139425009c4138e92766645dad7d0`.
+If that also checks, go on making a backup copy of your original `omcid` on the stick.
 
-Save it as `omcid_patch.base64`, then run:
 ```
```
Comment (simonebortolin) [inline _ont/ont-fs-com-gpon-onu-stick-with-mac.md:165]:
```suggestion
```sh
```

File: _ont/ont-fs-com-gpon-onu-stick-with-mac.md [hunk 5]
```diff
@@ -144,20 +144,32 @@ so you need to then modify the binary with the following hex patch which removes
...
-bspatch <your_original_omcid> omcid omcid.bspath
+cd /opt/lantiq/bin
+cp omcid omcid.original
 ```
 
-Now you have to copy via SCP the modified `omcid` binary in the `/opt/lantiq/bin/omcid` path, restart the stick and after that you can change the image version with the command:
+And finally to copy via SCP the modified `omcid` binary in the `/opt/lantiq/bin/omcid` path, restart the stick.
+After that you can change the image version with the command:
+
 ```
```
Comment (simonebortolin) [inline _ont/ont-fs-com-gpon-onu-stick-with-mac.md:173]:
```suggestion
```sh
```

**Review by simonebortolin — CHANGES_REQUESTED** [review]

File: _ont/ont-fs-com-gpon-onu-stick-with-mac.md [hunk 6]
```diff
@@ -144,25 +144,44 @@ so you need to then modify the binary with the following hex patch which removes
...
+This is the patch, encoded in base64
 ```
 QlNESUZGNDA1AAAAAAAAAD4AAAAAAAAA2C8JAAAAAABCWmg5MUFZJlNZYqnvBwAACFBSQWAAAMAA
 AAgAQCAAMQwIIwjImgDOdMvi7kinChIMVT3g4EJaaDkxQVkmU1lrJSbUAACFTAjAACAAAAiCAAAI
 IABQYAFKQ01INxUgd6Soj2JURm8pUR8XckU4UJBrJSbUQlpoORdyRThQkAAAAAA=
 ```
-
-Save it as `omcid_patch.base64`, then run:
+Save it on your computer (not on the stick) as `omcid_patch.base64`, then run:
+```shsh
```
Comment (simonebortolin) [inline _ont/ont-fs-com-gpon-onu-stick-with-mac.md:156]:
```suggestion
```sh
```

File: _ont/ont-fs-com-gpon-onu-stick-with-mac.md [hunk 7]
```diff
@@ -144,25 +144,44 @@ so you need to then modify the binary with the following hex patch which removes
...
+bspatch <your_original_omcid> omcid omcid.bspatch
 ```
-base64 -d omcid_patch.base64 > omcid.bspath
-bspatch <your_original_omcid> omcid omcid.bspath
+{% include alert.html content="if you don't have bspatch installed, most distributions includes it in bsdiff package" alert="Info" icon="svg-info" color="blue" %}
+
+After patching the resulting patched `omcid` should have an md5 checksum of `525139425009c4138e92766645dad7d0`.
+If that also checks, go on making a backup copy of your original `omcid` on the stick.
+
+```shsh
```
Comment (simonebortolin) [inline _ont/ont-fs-com-gpon-onu-stick-with-mac.md:165]:
```suggestion
```sh
```

**Review by simonebortolin — CHANGES_REQUESTED** [review]

File: _ont/ont-fs-com-gpon-onu-stick-with-mac.md [hunk 8]
```diff
@@ -144,25 +144,44 @@ so you need to then modify the binary with the following hex patch which removes
...
 ```
+
+Is also a good time to set the image0/image1_version. Crash has reported if they are not set correctly before reboot.
+```sh
 fw_setenv image0_version YOUR_IMAGE0_VERSION
 fw_setenv image1_version YOUR_IMAGE1_VERSION
 ```
 
+Now you can restart the stick.
+
```
Comment (simonebortolin) [inline _ont/ont-fs-com-gpon-onu-stick-with-mac.md:184]:
```suggestion
```

**Review by simonebortolin — CHANGES_REQUESTED** [review]

File: _ont/ont-fs-com-gpon-onu-stick-with-mac.md [hunk 9]
```diff
@@ -144,25 +144,44 @@ so you need to then modify the binary with the following hex patch which removes
...
 
-Now you have to copy via SCP the modified `omcid` binary in the `/opt/lantiq/bin/omcid` path, restart the stick and after that you can change the image version with the command:
+And finally to copy via SCP the modified `omcid` binary in the `/opt/lantiq/bin/omcid` path.
+Before restarting the stick and applying changes, make sure omcid has execution bit set.
+
+```sh
+chmod ugo+x /opt/lantiq/bin/omcid
 ```
+
+Is also a good time to set the image0/image1_version. Crash has reported if they are not set correctly before reboot.
```
Comment (ErnyTech) [inline _ont/ont-fs-com-gpon-onu-stick-with-mac.md:177]: You need to inform the user that you may need to specify the sw version again after reboot, omcid sometimes overwrites them when running without the patch

**Review by ErnyTech — CHANGES_REQUESTED** [review]

File: _ont/ont-fs-com-gpon-onu-stick-with-mac.md (hunk 9 above)
Comment (simonebortolin) [reply _ont/ont-fs-com-gpon-onu-stick-with-mac.md:177]: I would put it on an alertbox

**Review by simonebortolin — APPROVED** [review]

**Pull request merged**